File: first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        int[] letter = new int[26];
        // form counter map
        for (char c: s.toCharArray()) {
            letter[c-'a']++;
        }
        
        int index = 0;
        // find index of first unique character
        for (char c: s.toCharArray()) {
            if (letter[c-'a'] == 1) {
                return index;
            }
            index++;
        }

        // character not found, return zero
        return -1;
    }
}

File: 387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq_map = defaultdict(int)
        for c in s:
            freq_map[c] += 1
        
        for i, c in enumerate(s):
            if freq_map[c] == 1:
                return i

        return -1
